src/platform/mod.rs

```diff
@@ -8,12 +8,12 @@ mod platform;
 #[path = "macos/mod.rs"]
 mod platform;
 
-#[cfg(target_os = "linux")]
-#[path = "linux/mod.rs"]
+#[cfg(target_os = "unix")]
```

Comment:
You could change all these gates to: `#[cfg(all(unix, not(target_os = "macos")))]`. Because matching with all Unix platforms may conflict with MacOS.